backend/cn/main.ml

```diff
@@ -246,6 +234,35 @@ let main
       Cerb_debug.maybe_close_csv_timing_file_no_err ();
       Printexc.raise_with_backtrace exc (Printexc.get_raw_backtrace ())
 
+let generate_tests
+      (* Common *)
+      filename
+      macros
+      incl_dirs
+      incl_files
+      astprints
+      use_peval
+      no_inherit_loc
+      magic_comment_char_dollar
+      (* Test Generation *)
+      output_dir
+      max_unfolds
+      testing_framework
+  =
+  check_input_file filename;
+  let (prog4, (markers_env, ail_prog), _) =
+    handle_frontend_error
+      (frontend ~macros ~incl_dirs ~incl_files astprints ~do_peval:use_peval ~filename ~magic_comment_char_dollar)
+  in
+  let (_, sigma) = ail_prog in
+  let _ =
+    let open Resultat in
+    let@ prog5 = Core_to_mucore.normalise_file ~inherit_loc:(not(no_inherit_loc)) (markers_env, snd ail_prog) prog4 in
+    Cerb_colour.without_colour (fun () ->
+      TestGeneration.main ~output_dir ~filename ~max_unfolds sigma prog5 testing_framework)
+    ();
```

Comment:
Please run the specification well-formedness checking before you do test generation (see the main for the executable spec)
`let paused = Typing.run_to_pause Context.empty (Check.check_decls_lemmata_fun_specs prog5) in ...`

Reply:
Hmmm. Lift it out and move any flags it needs to `Common_flags`. You may wish to use the common `output-dir` flag for your test generation too. In fact, you may need to factor out everything up to that point, rather than just the handle error.

Reply:
Okay, I just print the message and point the user to `cn verify`